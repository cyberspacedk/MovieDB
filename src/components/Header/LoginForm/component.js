import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'formik';
import { Button } from 'antd';
import FormField from './FormField/component';

const LoginForm = ({ values, errors, touched }) => (
  <Form>
    <FormField
      type="text"
      name="username"
      placeHolder="Username"
      value={values}
      errors={errors}
      touched={touched}
    />
    <FormField
      type="password"
      name="password"
      placeHolder="Password"
      value={values}
      errors={errors}
      touched={touched}
    />
    <Button
      htmlType="submit"
      type="primary"
      disabled={
        (touched.username && errors.username) ||
        (touched.password && errors.password)
      }
    >
      Login
    </Button>
  </Form>
);

LoginForm.propTypes = {
  values: PropTypes.shape().isRequired,
  errors: PropTypes.shape().isRequired,
  touched: PropTypes.shape().isRequired,
};

export default LoginForm;
