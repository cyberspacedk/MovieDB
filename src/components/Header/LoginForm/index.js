import Loginform from './container';

export default Loginform;
