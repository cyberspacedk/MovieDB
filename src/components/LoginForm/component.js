/* eslint-disable no-unused-vars */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React from 'react';
import axios from 'axios';
import { Form, Icon, Input, Button } from 'antd';
import {
  REQUEST_TOKEN_PATH,
  GET_SESSION_ID_LOGIN_PATH,
  GET_SESSION_ID_PATH,
  DELETE_SESSION_ID_PATH,
} from '../../api/api';
import 'antd/dist/antd.css';

function hasErrors(fieldsError) {
  return Object.keys(fieldsError).some(field => fieldsError[field]);
}

class HorizontalLoginForm extends React.Component {
  componentDidMount() {
    this.props.form.validateFields();
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.refreshToken(REQUEST_TOKEN_PATH);

    this.props.form.validateFields((err, values) => {
      const { username, password } = values;
      const request_token = JSON.parse(localStorage.getItem('REQUEST_TOKEN'));

      this.props.auth(
        GET_SESSION_ID_LOGIN_PATH,
        username,
        password,
        request_token,
        GET_SESSION_ID_PATH,
      );
    });
  };

  handleLogout = e => {
    // e.preventDefault();
    const SID = JSON.parse(localStorage.getItem('SESSION_ID'));
    this.props.logout(DELETE_SESSION_ID_PATH, SID);
  };

  render() {
    const {
      getFieldDecorator,
      getFieldsError,
      getFieldError,
      isFieldTouched,
    } = this.props.form;

    // Only show error after a field is touched.
    const usernameError =
      isFieldTouched('username') && getFieldError('username');
    const passwordError =
      isFieldTouched('password') && getFieldError('password');

    const { isAuth } = this.props;
    console.log(isAuth);
    return (
      <Form layout="inline" onSubmit={this.handleSubmit}>
        {!isAuth ? (
          <>
            <Form.Item
              validateStatus={usernameError ? 'error' : ''}
              help={usernameError || ''}
            >
              {getFieldDecorator('username', {
                rules: [
                  { required: true, message: 'Please input your username!' },
                ],
              })(
                <Input
                  prefix={
                    <Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />
                  }
                  placeholder="Username"
                />,
              )}
            </Form.Item>

            <Form.Item
              validateStatus={passwordError ? 'error' : ''}
              help={passwordError || ''}
            >
              {getFieldDecorator('password', {
                rules: [
                  { required: true, message: 'Please input your Password!' },
                ],
              })(
                <Input
                  prefix={
                    <Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />
                  }
                  type="password"
                  placeholder="Password"
                />,
              )}
            </Form.Item>

            <Form.Item>
              <Button
                type="primary"
                htmlType="submit"
                disabled={hasErrors(getFieldsError())}
              >
                Log in
              </Button>
            </Form.Item>
          </>
        ) : (
          <Form.Item>
            <Button
              type="primary"
              htmlType="submit"
              onClick={this.handleLogout}
            >
              Log out
            </Button>
          </Form.Item>
        )}
      </Form>
    );
  }
}

const WrappedHorizontalLoginForm = Form.create({ name: 'horizontal_login' })(
  HorizontalLoginForm,
);

export default WrappedHorizontalLoginForm;
