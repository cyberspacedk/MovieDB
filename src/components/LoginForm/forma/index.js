import ContainerForm from './container';

export default ContainerForm;
