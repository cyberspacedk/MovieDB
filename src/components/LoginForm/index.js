import FormContainer from './container';

export default FormContainer;
