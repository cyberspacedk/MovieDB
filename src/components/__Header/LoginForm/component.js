import React from 'react';
import { Form, Field } from 'formik';
import FormField from './FormField/component';
import FormButton from './FormButton/component';

const LoginForm = () => (
  <Form>
    <Field name="username" component={FormField} />
    <Field name="password" component={FormField} />
    <Field component={FormButton} />
  </Form>
);

export default LoginForm;
