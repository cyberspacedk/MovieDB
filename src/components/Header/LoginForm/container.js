import { compose } from 'redux';
import { connect } from 'react-redux';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import { authUser } from '../../../store/authentifiction/actions';
import LoginForm from './component';

const mdtp = {
  authUser,
};

export const mapPropsToValues = () => ({
  username: '',
  password: '',
});

// РАЗОБРАТЬСЯ С ФУНКЦИЕЙ САБМИТА
export const handleSubmit = (values, { props, resetForm }) => {
  // eslint-disable-next-line no-shadow
  const { authUser } = props;
  const { rememberMe: _, ...user } = values;
  resetForm();
  authUser(user);
};

export const validationSchema = Yup.object().shape({
  username: Yup.string()
    .min(4, 'Минимум 4 символа')
    .matches(/^[a-z0-9_-]{4,16}$/, 'Запрещенные символы')
    .required('Обязательное поле'),
  password: Yup.string()
    .min(6, 'Минимум 6 символов')
    .required('Обязательное поле'),
});
export default compose(
  connect(
    null,
    mdtp,
  ),
  withFormik({
    mapPropsToValues,
    handleSubmit,
    validationSchema,
  }),
)(LoginForm);
