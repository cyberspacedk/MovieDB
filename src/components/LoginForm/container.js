/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  sendDataForCheck,
  userLogout,
  newRequestToken,
} from '../../Redux/Actions/setUserDataAction';
import {
  isAuthentificated,
  getUserLogin,
} from '../../Redux/Selectors/authReducer';
import LoginForm from './component';

const Header = styled.header`
  display: flex;
  background-color: #383838;
  padding: 10px 15px;
  margin-bottom: 20px;
  form {
    margin-left: auto;
  }
`;

const ContainerForm = ({
  checkToken,
  logoutUser,
  isAuth,
  refreshToken,
  userLogin,
}) => (
  <Header>
    <LoginForm
      auth={checkToken}
      logout={logoutUser}
      isAuth={isAuth}
      refreshToken={refreshToken}
      userLogin={userLogin}
    />
  </Header>
);
const mstp = state => ({
  isAuth: isAuthentificated(state),
  userLogin: getUserLogin(state),
});
const mdtp = {
  checkToken: sendDataForCheck,
  logoutUser: userLogout,
  refreshToken: newRequestToken,
};

export default connect(
  mstp,
  mdtp,
)(ContainerForm);
