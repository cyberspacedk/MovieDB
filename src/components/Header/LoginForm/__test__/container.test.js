/* eslint-disable no-unused-vars */
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import LoginForm, {
  mapPropsToValues,
  validationSchema,
  handleSubmit,
} from '../container';

configure({ adapter: new Adapter() });

describe('Container: LoginForm', () => {
  const props = {
    username: 'movie__watcher',
    sessionID: true,
  };

  const store = configureStore()({
    user: {
      username: 'movie__watcher',
      sessionId: '955360edb24b7e6d0179b7b4d6afdf5da2f56ada',
    },
  });

  const wrapper = mount(<LoginForm store={store} {...props} />);
  const loginForm = wrapper.find('LoginForm');

  it('Snapshot: should match', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Check: map dispatch to props', () => {
    expect(loginForm.props()).toEqual(
      expect.objectContaining({
        authUser: expect.any(Function),
      }),
    );
  });

  it('Check: mapPropsToValues function', () => {
    expect(mapPropsToValues()).toEqual({
      username: '',
      password: '',
    });
  });

  it('check validationSchema', () => {
    expect(validationSchema).toMatchSnapshot();
  });

  describe('check call submit func and inner function', () => {
    const values = {
      user: 'dfsf',
      password: 'fdfsf',
      rememberMe: true,
    };
    const { rememberMe: _x, ...userData } = values;
    const formikProps = {
      props: {
        authUser: jest.fn(),
      },
      resetForm: jest.fn(),
    };

    handleSubmit(userData, formikProps);

    it('should call resetForm func', () => {
      expect(formikProps.resetForm).toHaveBeenCalled();
    });

    it('should dispatch action with user info', () => {
      expect(formikProps.props.authUser).toHaveBeenCalledWith(userData);
    });
  });
});
