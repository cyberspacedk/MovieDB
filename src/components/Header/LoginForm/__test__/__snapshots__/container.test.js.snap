// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: LoginForm Snapshot: should match 1`] = `
<Connect(WithFormik(LoginForm))
  sessionID={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  username="movie__watcher"
>
  <WithFormik(LoginForm)
    authUser={[Function]}
    sessionID={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    username="movie__watcher"
  >
    <Formik
      authUser={[Function]}
      enableReinitialize={false}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "password": "",
          "username": "",
        }
      }
      isInitialValid={false}
      onSubmit={[Function]}
      render={[Function]}
      sessionID={true}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      username="movie__watcher"
      validateOnBlur={true}
      validateOnChange={true}
      validationSchema={[Function]}
    >
      <LoginForm
        authUser={[Function]}
        dirty={false}
        errors={Object {}}
        handleBlur={[Function]}
        handleChange={[Function]}
        handleReset={[Function]}
        handleSubmit={[Function]}
        initialValues={
          Object {
            "password": "",
            "username": "",
          }
        }
        isSubmitting={false}
        isValid={false}
        isValidating={false}
        registerField={[Function]}
        resetForm={[Function]}
        sessionID={true}
        setError={[Function]}
        setErrors={[Function]}
        setFieldError={[Function]}
        setFieldTouched={[Function]}
        setFieldValue={[Function]}
        setFormikState={[Function]}
        setStatus={[Function]}
        setSubmitting={[Function]}
        setTouched={[Function]}
        setValues={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        submitCount={0}
        submitForm={[Function]}
        touched={Object {}}
        unregisterField={[Function]}
        username="movie__watcher"
        validateField={[Function]}
        validateForm={[Function]}
        validateOnBlur={true}
        validateOnChange={true}
        values={
          Object {
            "password": "",
            "username": "",
          }
        }
      >
        <Form>
          <Component
            formik={
              Object {
                "dirty": false,
                "errors": Object {},
                "handleBlur": [Function],
                "handleChange": [Function],
                "handleReset": [Function],
                "handleSubmit": [Function],
                "initialValues": Object {
                  "password": "",
                  "username": "",
                },
                "isSubmitting": false,
                "isValid": false,
                "isValidating": false,
                "registerField": [Function],
                "resetForm": [Function],
                "setError": [Function],
                "setErrors": [Function],
                "setFieldError": [Function],
                "setFieldTouched": [Function],
                "setFieldValue": [Function],
                "setFormikState": [Function],
                "setStatus": [Function],
                "setSubmitting": [Function],
                "setTouched": [Function],
                "setValues": [Function],
                "status": undefined,
                "submitCount": 0,
                "submitForm": [Function],
                "touched": Object {},
                "unregisterField": [Function],
                "validate": undefined,
                "validateField": [Function],
                "validateForm": [Function],
                "validateOnBlur": true,
                "validateOnChange": true,
                "validationSchema": [Function],
                "values": Object {
                  "password": "",
                  "username": "",
                },
              }
            }
          >
            <form
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <FormikConnect(FieldInner)
                component={[Function]}
                name="username"
              >
                <FieldInner
                  component={[Function]}
                  formik={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialValues": Object {
                        "password": "",
                        "username": "",
                      },
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setError": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validate": undefined,
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validationSchema": [Function],
                      "values": Object {
                        "password": "",
                        "username": "",
                      },
                    }
                  }
                  name="username"
                >
                  <FormField
                    field={
                      Object {
                        "name": "username",
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": "",
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "password": "",
                          "username": "",
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "status": undefined,
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "password": "",
                          "username": "",
                        },
                      }
                    }
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bdVaJa",
                              "isStatic": true,
                              "lastClassName": "bBnrsp",
                              "rules": Array [
                                "
  position: relative;
  display: flex;
  justify-content: flex-end;
  input {
    margin-bottom: 3px;
  }
  [name='error-message'] {
    position: absolute;
    width: 100%;
    left: -100%;
    top: 0;
    color: #f70f16;
    font-size: 0.9rem;
  }
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "self": [Circular],
                            "styledComponentId": "sc-bdVaJa",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-bdVaJa bBnrsp"
                        >
                          <input
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Username"
                            type="text"
                            value=""
                          />
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </FormField>
                </FieldInner>
              </FormikConnect(FieldInner)>
              <FormikConnect(FieldInner)
                component={[Function]}
                name="password"
                type="password"
              >
                <FieldInner
                  component={[Function]}
                  formik={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialValues": Object {
                        "password": "",
                        "username": "",
                      },
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setError": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validate": undefined,
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validationSchema": [Function],
                      "values": Object {
                        "password": "",
                        "username": "",
                      },
                    }
                  }
                  name="password"
                  type="password"
                >
                  <FormField
                    field={
                      Object {
                        "name": "password",
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": "",
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "password": "",
                          "username": "",
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "status": undefined,
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "password": "",
                          "username": "",
                        },
                      }
                    }
                    type="password"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-bdVaJa",
                              "isStatic": true,
                              "lastClassName": "bBnrsp",
                              "rules": Array [
                                "
  position: relative;
  display: flex;
  justify-content: flex-end;
  input {
    margin-bottom: 3px;
  }
  [name='error-message'] {
    position: absolute;
    width: 100%;
    left: -100%;
    top: 0;
    color: #f70f16;
    font-size: 0.9rem;
  }
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "self": [Circular],
                            "styledComponentId": "sc-bdVaJa",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-bdVaJa bBnrsp"
                        >
                          <input
                            name="password"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Username"
                            type="password"
                            value=""
                          />
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </FormField>
                </FieldInner>
              </FormikConnect(FieldInner)>
              <FormikConnect(FieldInner)
                component={[Function]}
              >
                <FieldInner
                  component={[Function]}
                  formik={
                    Object {
                      "dirty": false,
                      "errors": Object {},
                      "handleBlur": [Function],
                      "handleChange": [Function],
                      "handleReset": [Function],
                      "handleSubmit": [Function],
                      "initialValues": Object {
                        "password": "",
                        "username": "",
                      },
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "registerField": [Function],
                      "resetForm": [Function],
                      "setError": [Function],
                      "setErrors": [Function],
                      "setFieldError": [Function],
                      "setFieldTouched": [Function],
                      "setFieldValue": [Function],
                      "setFormikState": [Function],
                      "setStatus": [Function],
                      "setSubmitting": [Function],
                      "setTouched": [Function],
                      "setValues": [Function],
                      "status": undefined,
                      "submitCount": 0,
                      "submitForm": [Function],
                      "touched": Object {},
                      "unregisterField": [Function],
                      "validate": undefined,
                      "validateField": [Function],
                      "validateForm": [Function],
                      "validateOnBlur": true,
                      "validateOnChange": true,
                      "validationSchema": [Function],
                      "values": Object {
                        "password": "",
                        "username": "",
                      },
                    }
                  }
                >
                  <FormButton
                    field={
                      Object {
                        "name": undefined,
                        "onBlur": [Function],
                        "onChange": [Function],
                        "value": Object {
                          "password": "",
                          "username": "",
                        },
                      }
                    }
                    form={
                      Object {
                        "dirty": false,
                        "errors": Object {},
                        "handleBlur": [Function],
                        "handleChange": [Function],
                        "handleReset": [Function],
                        "handleSubmit": [Function],
                        "initialValues": Object {
                          "password": "",
                          "username": "",
                        },
                        "isSubmitting": false,
                        "isValid": false,
                        "isValidating": false,
                        "registerField": [Function],
                        "resetForm": [Function],
                        "setError": [Function],
                        "setErrors": [Function],
                        "setFieldError": [Function],
                        "setFieldTouched": [Function],
                        "setFieldValue": [Function],
                        "setFormikState": [Function],
                        "setStatus": [Function],
                        "setSubmitting": [Function],
                        "setTouched": [Function],
                        "setValues": [Function],
                        "status": undefined,
                        "submitCount": 0,
                        "submitForm": [Function],
                        "touched": Object {},
                        "unregisterField": [Function],
                        "validateField": [Function],
                        "validateForm": [Function],
                        "validateOnBlur": true,
                        "validateOnChange": true,
                        "values": Object {
                          "password": "",
                          "username": "",
                        },
                      }
                    }
                  >
                    <Button
                      block={false}
                      ghost={false}
                      htmlType="submit"
                      loading={false}
                      type="primary"
                    >
                      <Wave>
                        <button
                          className="ant-btn ant-btn-primary"
                          onClick={[Function]}
                          type="submit"
                        >
                          <span
                            key=".0"
                          >
                            Login
                          </span>
                        </button>
                      </Wave>
                    </Button>
                  </FormButton>
                </FieldInner>
              </FormikConnect(FieldInner)>
            </form>
          </Component>
        </Form>
      </LoginForm>
    </Formik>
  </WithFormik(LoginForm)>
</Connect(WithFormik(LoginForm))>
`;

exports[`Container: LoginForm check validationSchema 1`] = `
ObjectSchema {
  "_blacklist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "_conditions": Array [],
  "_defaultDefault": [Function],
  "_deps": Array [],
  "_excludedEdges": Array [],
  "_exclusive": Object {},
  "_mutate": undefined,
  "_nodes": Array [
    "password",
    "username",
  ],
  "_options": Object {
    "abortEarly": true,
    "recursive": true,
  },
  "_type": "object",
  "_typeError": [Function],
  "_whitelist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "fields": Object {
    "password": StringSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_deps": Array [],
      "_exclusive": Object {
        "min": true,
        "required": true,
      },
      "_mutate": undefined,
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "string",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "tests": Array [
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
    },
    "username": StringSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_deps": Array [],
      "_exclusive": Object {
        "min": true,
        "required": true,
        "undefined": false,
      },
      "_mutate": undefined,
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "string",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
    },
  },
  "tests": Array [],
  "transforms": Array [
    [Function],
  ],
}
`;
